web/src/user_profile.js

```diff
@@ -803,26 +803,30 @@ export function initialize() {
         e.preventDefault();
         const stream_id = Number.parseInt(user_profile_subscribe_widget.value(), 10);
         const sub = sub_store.get(stream_id);
+
+        if (!sub) {
+            return;
+        }
+
         const target_user_id = Number.parseInt($("#user-profile-modal").attr("data-user-id"), 10);
         const $alert_box = $("#user-profile-streams-tab .stream_list_info");
+
```

Comment:
Adding or removing lines unnecessarily should be avoided.

Reply:
One other thing to note is that the success message should be near the button. 

One way to solve that might be to put the subscribe button under the dropdown, and then show the message to the right of the button—there should be space enough in that case for internationalization. It would be good to see all of this in action at mobile-scale viewports, too.